app/src/main/java/free/rm/skytube/gui/fragments/YouTubePlayerV2Fragment.java

```diff
@@ -151,982 +162,1041 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
+        Context context = getContext();
+        DefaultRenderersFactory defaultRenderersFactory = new DefaultRenderersFactory(context);
+
+        return ExoPlayerFactory.newSimpleInstance(getContext(), defaultRenderersFactory, trackSelector, new DefaultLoadControl(), null, bandwidthMeter);
+    }
+
+
+    /**
+     * Will setup the HUD's details according to the contents of {@link #youTubeVideo}.  Then it
+     * will try to load and play the video.
+     */
+    private void setUpHUDAndPlayVideo() {
+        setupInfoDisplay(youTubeVideo);
+
+        new ResumeVideoTask(getContext(), youTubeVideo.getId(), position -> {
+            playerInitialPosition = position;
+            YouTubePlayerV2Fragment.this.loadVideo();
+        }).ask();
+    }
+
+    private void setupInfoDisplay(YouTubeVideo video) {
+        getSupportActionBar().setTitle(video.getTitle());
+        videoDescriptionBinding.videoDescTitle.setText(video.getTitle());
+        videoDescriptionBinding.videoDescChannel.setText(video.getChannelName());
+        videoDescriptionBinding.videoDescViews.setText(video.getViewsCount());
+        videoDescriptionBinding.videoDescPublishDate.setText(video.getPublishDatePretty());
+
+        if (video.getDescription() != null) {
+            Linker.setTextAndLinkify(videoDescriptionBinding.videoDescDescription, video.getDescription());
+        }
+        final boolean hasLikes = video.getLikeCountNumber() != null;
+        setTextAndVisibility(videoDescriptionBinding.videoDescLikes, hasLikes, video.getLikeCount());
+        final boolean hasDislikes = video.getDislikeCountNumber() != null;
+        setTextAndVisibility(videoDescriptionBinding.videoDescDislikes, hasDislikes, video.getDislikeCount());
+        setValueAndVisibility(videoDescriptionBinding.videoDescLikesBar, video.isThumbsUpPercentageSet(), video.getThumbsUpPercentage());
+        setVisibility(videoDescriptionBinding.videoDescRatingsDisabled, !hasLikes && !hasDislikes);
+
+        if (SBVideoInfo.isLoaded()) {
+            Log.d(TAG, "SBInfo has loaded");
+            Handler handler = new Handler(Looper.getMainLooper());
+            for (SBSegment segment : SBVideoInfo.getSegments()) {
+                long startPosMs = Double.valueOf(segment.getStartPos() * 1000).longValue();
+
+                player.createMessage((messageType, payload) -> {
+                    SBSegment payloadSegment = (SBSegment) payload;
+
+                    handler.post(new Runnable() {
+                        public void run() {
+                            String rawCategory = payloadSegment.getCategory();
+                            String categoryLabel = rawCategory;
+
+                            try {
+                                Field resourceField = R.string.class.getDeclaredField("sponsorblock_category_" + rawCategory);
+                                int resourceId = resourceField.getInt(resourceField);
+                                categoryLabel = getString(resourceId);
```

Comment:
I don't think using reflection is the best solution.
Maybe, you can add a Map<String, Integer> to SBTasks, which maps from category names to resourceIds.
And you can even use that mapping in SBTimeBarView.initDraw to initialize the colors

Reply:
I tried that first, but it was a lot of similar code and looked quite ugly. Reflection was the cleanest solution, and it uses a catch with a fallback in case this is incorrect.